00_basics/sc_dimensions.ipynb

```diff
@@ -0,0 +1,771 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "aerial-emperor",
+   "metadata": {},
+   "source": [
+    "### Vector\n",
+    "\n",
+    "We can store multiple scalar values into an [Array](https://doc.sccode.org/Classes/Array.html).\n",
+    "Have in mind that [indexing of arrays start at 0 in most programming languages](https://www.cs.utexas.edu/users/EWD/transcriptions/EWD08xx/EWD831.html) where in math the convention is to start indexing from 1."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "convenient-equity",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "-> [ 2, 3 ]"
+     ]
+    }
+   ],
+   "source": [
+    "v = [2, 3];\n",
+    "v;"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "governmental-watch",
+   "metadata": {},
+   "source": [
+    "We can scale each element of our array with a scalar by multiplying a scalar with our array `v`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "alpha-extent",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "-> [ 8, 12 ]"
+     ]
+    }
+   ],
+   "source": [
+    "v*4"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "interested-variety",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "-> [ 0.5, 0.75 ]"
+     ]
+    }
+   ],
+   "source": [
+    "v/4"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "manual-valuable",
+   "metadata": {},
+   "source": [
+    "We can also multiply the array with another array."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "advance-wesley",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "-> [ 8, 27 ]"
+     ]
+    }
+   ],
+   "source": [
+    "[2,3]*[4,9]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "electoral-eagle",
+   "metadata": {},
+   "source": [
+    "Lets see what happens when we multiply non-matching dimensions of our arrays."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "humanitarian-robin",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "-> [ 8, 15, 12 ]"
+     ]
+    }
+   ],
+   "source": [
+    "[2,3]*[4,5,6]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "considerable-might",
+   "metadata": {},
+   "source": [
+    "A bit unexpecting that we do not receive an error - instead SuperCollider tries to fix our error here by cycling the smaller array until we have gone through all arrays once fully.\n",
```

Comment:
It shouldn't be considered "our error" which is "fixed", but we need to explain why this is intended and useful behaviour. 

> "SuperCollider expands arrays in such operations to the longest array and cycles through the shorter ones. This allows for very concise code, but you have to take care about the input arrays yourself."

we could add that there are adverbs for controlling this, but maybe this is overkill? `[2,3] *.s [4,5,6]` e.g. will do the shortest.

It is a good idea actually to have an adverb for throwing an error, we should add that …

Reply:
Good point - added it